### ISSUE TITLE
'ClientCapabilityToken' object has no attribute 'generator'

### ISSUE THREAD

**Issue by alidabour** [TwilioDevEd/browser-calls-django#10]
I did install the requiremnt and python version is 2.7.12 
But I found this error and I can't connect to twilio

**Comment by quentusrex** [comment]
Same, and here is the backtrace:
127.0.0.1 - - [06/Jan/2018 23:50:22] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [06/Jan/2018 23:50:22] "GET /support/token?forPage=%2F HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/src/browser-calls-flask/venv/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/src/browser-calls-flask/browser_calls_flask/views.py", line 56, in get_token
    token = capability.generate()
AttributeError: 'ClientCapabilityToken' object has no attribute 'generate'

**Comment by alidabour** [comment]
This worked And I connected to twilio successfully but I could not call any one, The call closed without getting any error messages.
```
capability = ClientCapabilityToken(settings.TWILIO_ACCOUNT_SID,settings.TWILIO_AUTH_TOKEN,ttl=xxx)
token = capability.to_jwt()

```

**Comment by akshatpv** [comment]
@alidabour Where did you include this? And did you get it working eventually? Thanks!

**Comment by alidabour** [comment]
> @alidabour Where did you include this? And did you get it working eventually? Thanks!

I can't remember as it has been a long time since I tried Twilio.

**Comment by akshatpv** [comment]
I see. Thanks! In case you get to know anything do let me know.

**Comment by smendes** [comment]
Hello everyone,
I apologize for the time it took to answer this issue, that tutorial was indeed very outdated compared to the changes in the libraries.

I've upgraded the tutorial to make it work with the latest versions of all dependencies. I've also extended the README to add basics use cases for this tutorial.

The reason why your call dropped immediately is because  the "support agent" needs to have a tab open on `http://localhost:8000/support/dashboard`.
